src/compiler/crystal/tools/doc/project_info.cr

```diff
@@ -1,19 +1,29 @@
 module Crystal::Doc
-  record ProjectInfo, name : String, version : String do
-    def self.new_with_defaults(name, version)
-      version ||= find_git_version
+  class ProjectInfo
+    property! name : String
+    property! version : String
 
-      unless name && version
-        shard_name, shard_version = read_shard_properties
-        name ||= shard_name
-        version ||= shard_version
+    def initialize(@name : String? = nil, @version : String? = nil )
+    end
 
-        unless name && version
-          return yield name, version
+    def_equals_and_hash @name, @version
+
+    def fill_with_defaults
+      unless self.version
```

Comment:
Shouldn't this use @ivar? Calling bang getter will raise in case of `nil`.

ditto below